omod/src/test/java/org/openmrs/module/owa/web/controller/OwaRestControllerTest.java

```diff
@@ -138,4 +138,15 @@ public void install_wrongDownloadUrl() throws Exception {
 		List<App> appList = controller.install(requestData, request, response);
 		Assert.assertEquals("owa.not_a_zip", request.getSession().getAttribute(WebConstants.OPENMRS_ERROR_ATTR));
 	}
+	
+	@Test
+	public void install_rightDownloadUrl_withFileName() throws Exception {
+		HttpServletRequest request = new MockHttpServletRequest(new MockServletContext(), "POST", "/rest/owa/installapp");
+		HttpServletResponse response = new MockHttpServletResponse();
+		String downloadUrl = "https://bintray.com/openmrs/owa/download_file?file_path=cohortbuilder-1.0.0-beta.zip";
+		OwaRestController controller = (OwaRestController) applicationContext.getBean("owaRestController");
```

Comment:
Does this test pass without an internet connection?

Reply:
@dkayiwa No, the test needs internet for the downloading of the data stream from bintary and I believe mocking that process would defeat the essence of the test as one of it's crucial functionality is extracting the right data from the bintary host.

Reply:
@tolupatrick004 we do not encourage tests that require an internet connection